aten/src/ATen/native/cuda/Bucketization.cu

```diff
@@ -122,7 +129,10 @@ Tensor& searchsorted_out_cuda(const Tensor& sorted_sequence, const Tensor& self,
   searchsorted_maybe_trim_input_tensors(trimmed_input, trimmed_boundaries, self, sorted_sequence);
   const Tensor& final_input = trimmed_input.defined() ? trimmed_input : self;
   const Tensor& final_boundaries = trimmed_boundaries.defined() ? trimmed_boundaries : sorted_sequence;
-  dispatch(result, final_input, final_boundaries, out_int32, right);
+  dispatch(out, final_input, final_boundaries, out_int32, right);
+  if (!result.is_contiguous()) {
+    result.copy_(out);
+  }
```

Comment:
Same as above, could you combine the two dispatch cases in an if-else and have a single is_contiguous check?